sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+// temperatures.
+static az_result invoke_getMaxMinReport(az_span payload, az_span response, az_span* out_response)
+{
+  // Parse the "since" field in the payload.
+  az_json_parser jp;
+  AZ_RETURN_IF_FAILED(az_json_parser_init(&jp, payload, NULL));
+  AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+  if (jp.token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
+  }
+
+  // Get the user specified "since". Note that we don't honor this option to simplify the sample. It
+  // is merely passed on to the method call and return payload.
+  az_span start_time_span = AZ_SPAN_NULL;
+  while (jp.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+  {
+    if (az_json_token_is_text_equal(&jp.token, report_command_payload_value_span))
+    {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+      int32_t incoming_since_value_len;
+      AZ_RETURN_IF_FAILED(az_json_token_get_string(
+          &jp.token,
+          incoming_since_value,
+          sizeof(incoming_since_value),
+          &incoming_since_value_len));
+      start_time_span = az_span_init((uint8_t*)incoming_since_value, incoming_since_value_len);
+      break;
+    }
+
+    // else ignore token.
+    AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+  }
+
+  // Set the response payload to error if the "since" field was not sent
+  if (az_span_ptr(start_time_span) == NULL)
+  {
+    response = report_error_payload;
+    return AZ_ERROR_ITEM_NOT_FOUND;
+  }
+
+  // Get the current time as a string
+  time_t rawtime;
+  struct tm* timeinfo;
+  time(&rawtime);
+  timeinfo = localtime(&rawtime);
+  size_t len = strftime(end_time_buffer, sizeof(end_time_buffer), iso_spec_time_format, timeinfo);
+  az_span end_time_span = az_span_init((uint8_t*)end_time_buffer, (int32_t)len);
+
+  az_json_builder json_builder;
+  AZ_RETURN_IF_FAILED(az_json_builder_init(&json_builder, response, NULL));
+  AZ_RETURN_IF_FAILED(
+      build_command_response_payload(&json_builder, start_time_span, end_time_span, out_response));
+
+  return AZ_OK;
+}
+
+// Send the response of the command invocation
+static int send_command_response(
+    az_iot_hub_client_method_request* request,
+    uint16_t status,
+    az_span response)
+{
+  int rc;
+  // Get the response topic to publish the command response
+  if (az_failed(
+          rc = az_iot_hub_client_methods_response_get_publish_topic(
+              &client,
+              request->request_id,
+              status,
+              commands_response_topic,
+              sizeof(commands_response_topic),
+              NULL)))
+  {
+    printf("Unable to get twin document publish topic\n");
+    return rc;
+  }
+
+  printf("Status: %u\tPayload:", status);
+  char* payload_char = (char*)az_span_ptr(response);
+  if (payload_char != NULL)
+  {
+    for (int32_t i = 0; i < az_span_size(response); i++)
+    {
+      putchar(*(payload_char + i));
```

Comment:
>putchar [](start = 6, length = 7)

I don't know enough about new SDK so maybe this is way to do it, but do we have to do a putchar() loop here?  Is there anyway to do a printf() directly from command_response_span, like with a "%.s" identifier?

Reply:
The original intent was to remove "fancy" printf features to allow for smaller/limited versions of printf to work. Considering this sample is for windows and linux only, we can revisit that.